### ISSUE TITLE
Inefficiency when both div_ and rem_ are needed

### ISSUE THREAD

**Issue by spj101** [vermaseren/form#282]
There is an inefficiency when the user wants the result of both div_ and rem_. These are computed simultaneously by the `divmod` algorithms inside FORM but then only one is returned. This effectively doubles the time the user must wait.

Is it possible to add something like the following?
```
L q,r = divrem_(f,g);
```
or
```
divrem(f,g,q,r);
```

Thanks!

**Comment by tueda** [comment]
I really like the first syntax in which `divrem_` returns a tuple, but probably FORM never gets such a cozy syntax. I'm not sure about the exact meaning of the second one, but I guess `divrem_(f,g,q,r)` that returns a product of the quotient and reminder that are hold in functions `q` and `r`, so `q(...)*r(...)`. This would be fine but at some point one will encounter the MaxTermSize problem.

**Comment by vermaseren** [comment]
I guess the proper syntax would be something like
   divrem_(f,g,$remainder)
where the function returns the quotient and the remainder gets stored in the dollar variable.
One can then also think of remdiv_ that returns the remainder and puts the quotient in the $.
The only little complication is thatthis function needs a special status internally to prevent the
$ from being expanded. Things like this have been done already with some other functions
and names of expressions and dollars.
Actually div_ and rem_ also do not expand expressions until they get busy. Hence I think it
should not be much work.

Jos

**Comment by vermaseren** [comment]
Well, if you are thinking that big….
In that case you can already use
L  F = div_(f,g);
.sort
L G = f-g*F;   * remainder.

I just checked. It is not quite easy to extract both from the current routines.
The routine is poly_divmod, but there parts where it assumes that it is calculating only one of the two.
The remaining part (for either syntax) is the easy part. 

Jos

**Comment by spj101** [comment]
I agree, we can of course we can compute using a multiplication rather than calling `rem_`. This is slightly less efficient than if the function just returned the sorted(?) remainder which is already known to it. 

Yes I see that `poly_divmod` only converts either the quotient or the remainder from the C++ poly class to the syntax needed by the C part of FORM. The thing I would be least sure how to implement is how to return an additional result in a dollar variable (or other expression).
